api/base/serializers.py

```diff
@@ -328,6 +350,26 @@ def __init__(self, **kwargs):
         self.link_type = kwargs.pop('link_type', 'url')
         super(JSONAPIHyperlinkedGuidRelatedField, self).__init__(read_only=True, **kwargs)
 
+    def resolve(self, resource):
+        """
+        Resolves the view for target node or target comment when embedding.
+        """
+        embed_value = resource.target._id
+        if resource.target._name == 'node':
+            view_name = 'nodes:node-detail'
+            lookup_url_kwarg = 'node_id'
+        else:
```

Comment:
What kind of error were you thinking?  If I want to raise an error if it's not node or comments, seems like the entire field should be for resolving the target on comments.   Change from JSONAPIHyperlinkedGuidRelatedField to TargetField or something of the sort.